libres/lib/include/ert/enkf/local_ministep.hpp

```diff
@@ -34,16 +30,20 @@ extern "C" {
 #include <ert/enkf/local_obsdata.hpp>
 #include <ert/enkf/local_obsdata_node.hpp>
 #include <ert/enkf/obs_data.hpp>
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 typedef struct local_ministep_struct local_ministep_type;
 
 local_ministep_type *
 local_ministep_alloc(const char *name, analysis_module_type *analysis_module);
 void local_ministep_free(local_ministep_type *ministep);
 void local_ministep_free__(void *arg);
-hash_iter_type *
-local_ministep_alloc_dataset_iter(const local_ministep_type *ministep);
+
 const char *local_ministep_get_name(const local_ministep_type *ministep);
+std::unordered_map<std::string, const local_dataset_type *>
+local_ministep_get_datasets(const local_ministep_type *ministep);
```

Comment:
Add a bit of protection so that these get name-mangled and cannot be used from cwrap.

```suggestion
extern "C++" std::unordered_map<std::string, const local_dataset_type *>
local_ministep_get_datasets(const local_ministep_type *ministep);
```

Reply:
Don't they get name-mangled by default?